internal/connectors/plugins/public/mangopay/accounts.go

```diff
@@ -50,55 +52,27 @@ func (p Plugin) fetchNextAccounts(ctx context.Context, req models.FetchNextAccou
 			return models.FetchNextAccountsResponse{}, err
 		}
 
-		if len(pagedAccounts) == 0 {
-			break
-		}
-
-		for _, account := range pagedAccounts {
-			accountCreationDate := time.Unix(account.CreationDate, 0)
-			switch accountCreationDate.Compare(oldState.LastCreationDate) {
-			case -1, 0:
-				// creationDate <= state.LastCreationDate, nothing to do,
-				// we already processed this account.
-				continue
-			default:
-			}
-
-			raw, err := json.Marshal(account)
-			if err != nil {
-				return models.FetchNextAccountsResponse{}, err
-			}
-
-			accounts = append(accounts, models.PSPAccount{
-				Reference:    account.ID,
-				CreatedAt:    accountCreationDate,
-				Name:         &account.Description,
-				DefaultAsset: pointer.For(currency.FormatAsset(supportedCurrenciesWithDecimal, account.Currency)),
-				Metadata: map[string]string{
-					userIDMetadataKey: from.ID,
-				},
-				Raw: raw,
-			})
-
-			newState.LastCreationDate = accountCreationDate
-
-			if len(accounts) >= req.PageSize {
-				break
-			}
-		}
-
-		if len(pagedAccounts) < req.PageSize {
-			break
+		accounts, err = fillAccounts(pagedAccounts, accounts, from, oldState)
+		if err != nil {
+			return models.FetchNextAccountsResponse{}, err
 		}
 
-		if len(accounts) >= req.PageSize {
-			hasMore = true
+		needMore, hasMore = pagination.ShouldFetchMore(accounts, pagedAccounts, req.PageSize)
+		if !needMore || !hasMore {
 			break
 		}
 
 		page++
 	}
 
+	if !needMore {
+		accounts = accounts[:req.PageSize]
+	}
+
+	if len(accounts) > 0 {
+		newState.LastCreationDate = accounts[len(accounts)-1].CreatedAt
+	}
```

Comment:
`accounts[:req.PageSize]` will result in a panic if `len(accounts) == 0` so let's put our `needsMore` condition inside the if block on line 72

Reply:
It's not possible to have `len(accounts) == 0` if we are on the !needMore case